recipes/doctest/2.3.5/conanfile.py

```diff
@@ -24,11 +24,22 @@ def package(self):
         self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
         self.copy(pattern="*doctest.h", dst="include", src=self._source_subfolder)
 
+        cmake_script_dirs = {
+            "src": os.path.join(self._source_subfolder, "scripts/cmake"),
+            "dst": "lib/cmake"
+        }
+
+        self.copy(pattern="doctest.cmake", **cmake_script_dirs)
+        self.copy(pattern="doctestAddTests.cmake", **cmake_script_dirs)
+
     def package_info(self):
         if self._is_mingw:
             # See https://sourceforge.net/p/mingw-w64/bugs/727/
             # can't use destructors in thread_local with mingw
             self.cpp_info.defines.append("DOCTEST_THREAD_LOCAL=")
 
+        self.cpp_info.builddirs.append("lib/cmake")
+        self.cpp_info.build_modules.append("lib/cmake/doctest.cmake")
+
```

Comment:
Why did you add `doctestAddTests.cmake` if it's not listed in `self.cpp_info.build_modules` ?

Reply:
`doctestAddTests.cmake` is not meant to be included directly. `doctest.cmake` adds a `POST_BUILD` command, which calls it to discover tests (it's happening in [here](https://github.com/onqtam/doctest/blob/503de03ee2cbbfc58c2063dce36dc96f917f2da3/scripts/cmake/doctest.cmake#L138), `_DOCTEST_DISCOVER_TESTS_SCRIPT` points to the location of `doctestAddTests.cmake`).

Reply:
thanks for answering, so we are good!